src/v/archival/ntp_archiver_service.cc

```diff
@@ -1531,7 +1531,8 @@ ss::future<ntp_archiver::manifest_updated> ntp_archiver::apply_retention() {
     }
 
     auto next_start_offset = retention_calculator->next_start_offset();
-    if (next_start_offset) {
+    auto current_start_offset = manifest().get_start_offset();
+    if (next_start_offset && next_start_offset > current_start_offset) {
```

Comment:
I think this addresses the same problem as https://github.com/redpanda-data/redpanda/pull/9494

With that fix in, the new leader will sync and apply truncations from previous terms, and the new leader will pick a new offset based on a fresh manifest that includes those truncations

This is probably a good safeguard, though a downside is that it could hide those kinds of issues

Reply:
Yeah, that's a good point. I completely blanked out on the existence of #9494. I'm good with merging that and skipping my change if you think it's better for observability.

Reply:
Yeah let's just leave it for now -- I imagine whatever else it uncovers will be actual bugs worth fixing.

Just merged #9494